keras/backend/cntk_backend.py

```diff
@@ -2750,8 +2751,33 @@ def cumprod(x, axis=0):
     raise NotImplementedError
 
 
-def arange(start, stop=None, step=1, dtype='int32'):
-    raise NotImplementedError
+def arange(start, stop=None, step=1, dtype='float32'):
+    """Creates a 1D tensor containing a sequence of integers.
+    The function arguments use the same convention as
+    Theano's arange: if only one argument is provided,
+    it is in fact the "stop" argument and "start" is 0.
+    The default type of the returned tensor is `'float32'` to
+    match CNTK 's default.
+    # Arguments
+        start: Start value.
+        stop: Stop value.
+        step: Difference between two successive values.
+        dtype: Integer dtype to use.
+    # Returns
+        An integer tensor.
+    """
+    # Match the behavior of numpy and Theano by returning an empty sequence.
+    if stop is None:
+        try:
+            if start < 0:
+                start = 0
+        except TypeError:
+            # Handle case where start is a tensor
+            start = 0 if start.value < 0 else start.value
+
+    result = C.variables.Constant(np.arange(start, stop, step),
```

Comment:
Thank you for the PR, @SriRangaTarun. The following will work:
```python
ctype = _convert_string_dtype(dtype)
return variable(value=np.arange(start, stop, step).astype(ctype), dtype=dtype, name=name)
```

Reply:
Fixed